src/bg_daemon/fetchers/imgurfetcher.py

```diff
@@ -160,6 +164,9 @@ def fetch(self, imgobject, filename):
 
         req = requests.get(imgobject.link)
 
+        if not isinstance(req, requests.Response):
```

Comment:
Why are you checking if the returned object is indeed a response? Are you trying to find out if it's a good return code or why are you doing this? Instead of having to check yourself, you can use this handy method:

```python
req.raise_for_status()
```